specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/artifact.json

```diff
@@ -0,0 +1,1615 @@
+        "tags": [
+          "Artifact"
+        ],
+        "summary": "Create a batch of empty Artifacts.",
+        "description": "Create a Batch of empty Artifacts from the supplied paths.",
+        "operationId": "Artifacts_BatchCreateEmptyArtifacts",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/workspaceNameParameter"
+          },
+          {
+            "name": "origin",
+            "in": "path",
+            "description": "The origin of the Artifact.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "container",
+            "in": "path",
+            "description": "The container name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "artifactPaths",
+            "in": "body",
+            "description": "The list of Artifact paths to create.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ArtifactPathList"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Batch of empty Artifacts created successfully.",
+            "schema": {
+              "$ref": "#/definitions/BatchArtifactContentInformationResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Artifact"
+        ],
+        "summary": "Delete Batch of Artifact Metadata.",
+        "description": "Delete a Batch of Artifact Metadata.",
+        "operationId": "Artifacts_BatchDeleteMetaData",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/workspaceNameParameter"
+          },
+          {
+            "name": "origin",
+            "in": "path",
+            "description": "The origin of the Artifact.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "container",
+            "in": "path",
+            "description": "The container name.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "artifactPaths",
```

Comment:
A `DELETE` method with a request body is suspect. While the HTTP spec doesn't explicitly say that it is forbidden, it is implied that since there is no defined semantics for the body, services should ignore it.

Reply:
This is a case of Batch Delete which requires a list of items to be deleted. Weather its alist of strings or list of objects, I dont see a better way to call the operation and pass the list. Any suggestions ?

Reply:
I would expect a batch operation to be a POST. For DELETE, like I mentioned above, request bodies are supposed to be ignored by the service, and the request URI is supposed to identify the resource to delete. Neither of those assumptions are true here.

Is this an idempotent operation? Is it an atomic operation?

Reply:
This operation is idempotent but not atomic. This API is an "in-the-wild" API, not sure if we can remove it, but we could make a new version and document that one. 

I don't think we'd have to be able to use these same routes, so might have to come up  with some POST /artifacts/batch:delete type thing, since i think there is already a POST /artifacts/batch that creates artifacts

Reply:
Is that better?

Reply:
Thanks I @andscho-msft, I will make the changes.

Reply:
fix on the way